x/oracle/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,132 @@
+package keeper_test
+
+import (
+	"context"
+	"math/rand"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/umee-network/umee/x/oracle/types"
+)
+
+func (s *IntegrationTestSuite) TestQuerier_ActiveExchangeRates() {
+	const (
+		exchangeRate = "umee"
+	)
+	app, ctx := s.app, s.ctx
+
+	app.OracleKeeper.SetExchangeRate(ctx, exchangeRate, sdk.OneDec())
+	_, err := s.queryClient.ActiveExchangeRates(context.Background(), &types.QueryActiveExchangeRatesRequest{})
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TestQuerier_ExchangeRates() {
+	const (
+		exchangeRate      = "umee"
+		exchangeRateDenom = "uumee"
+	)
+	app, ctx := s.app, s.ctx
+
+	app.OracleKeeper.SetExchangeRate(ctx, exchangeRate, sdk.OneDec())
+	res, err := s.queryClient.ExchangeRates(context.Background(), &types.QueryExchangeRatesRequest{
+		Denom: exchangeRateDenom,
+	})
+	s.Require().NoError(err)
+	s.Require().Equal(sdk.DecCoins{
+		sdk.NewDecCoinFromDec(exchangeRateDenom, sdk.OneDec()),
+	}, res.ExchangeRates)
+}
+
+func (s *IntegrationTestSuite) TestQuerier_FeeederDelegation() {
+	app, ctx := s.app, s.ctx
+
+	feederAddr := sdk.AccAddress([]byte("addr________________"))
+	feederAcc := app.AccountKeeper.NewAccountWithAddress(ctx, feederAddr)
+	app.AccountKeeper.SetAccount(ctx, feederAcc)
+
+	err := s.app.OracleKeeper.ValidateFeeder(ctx, feederAddr, valAddr)
+	s.Require().Error(err)
+
+	s.app.OracleKeeper.SetFeederDelegation(ctx, valAddr, feederAddr)
+
+	err = s.app.OracleKeeper.ValidateFeeder(ctx, feederAddr, valAddr)
+	s.Require().NoError(err)
+
+	_, err = s.queryClient.FeederDelegation(context.Background(), &types.QueryFeederDelegationRequest{
+		ValidatorAddr: valAddr.String(),
+	})
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TestQuerier_MissCounter() {
+	app, ctx := s.app, s.ctx
+	missCounter := uint64(rand.Intn(100))
+
+	res, err := s.queryClient.MissCounter(context.Background(), &types.QueryMissCounterRequest{
+		ValidatorAddr: valAddr.String(),
+	})
+	s.Require().NoError(err)
+	s.Require().Equal(res.MissCounter, uint64(0))
+
+	app.OracleKeeper.SetMissCounter(ctx, valAddr, missCounter)
+
+	res, err = s.queryClient.MissCounter(context.Background(), &types.QueryMissCounterRequest{
+		ValidatorAddr: valAddr.String(),
+	})
+	s.Require().NoError(err)
+	s.Require().Equal(res.MissCounter, missCounter)
+}
+
+func (s *IntegrationTestSuite) TestQuerier_AggregatePrevote() {
+	app, ctx := s.app, s.ctx
+
+	prevote := types.AggregateExchangeRatePrevote{
+		Hash:        "hash",
+		Voter:       addr.String(),
+		SubmitBlock: 0,
+	}
+	app.OracleKeeper.SetAggregateExchangeRatePrevote(ctx, valAddr, prevote)
+
+	_, err := app.OracleKeeper.GetAggregateExchangeRatePrevote(ctx, valAddr)
+	s.Require().NoError(err)
+
+	_, err = s.queryClient.AggregatePrevote(context.Background(), &types.QueryAggregatePrevoteRequest{
+		ValidatorAddr: valAddr.String(),
+	})
+	s.Require().NoError(err)
```

Comment:
Consider checking the `_` vars their expected value, unless you deem that to be redundant.

Reply:
Agree here 👍